File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> counter; 
        for(int i = 0; i < nums.size(); i++){
            counter[nums[i]]++; 
            if (counter[nums[i]] > 1)return true; 
        }
        return false; 
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # Naive approach, sort and return the longest length. 
        if (len(nums) == 0): return 0
        nums.sort() 
        max_length, curr_length = 1, 1 
        for i in range(1, len(nums)): 
            if (nums[i] - nums[i - 1] == 1): 
                curr_length += 1
                max_length = max(max_length, curr_length)
            elif (nums[i] == nums[i - 1]): 
                continue
            else:  
                curr_length = 1
        return max_length